portality/dao.py

```diff
@@ -685,7 +886,20 @@ def dump(cls, q=None, page_size=1000, limit=None, out=None, out_template=None, o
             out = sys.stdout
 
         count = 0
-        for record in cls.scroll(q, page_size=page_size, limit=limit, wrap=False, keepalive=scroll_keepalive):
+
+        iterator = None
+        if iterate_method == "unstable":
```

Comment:
Since we've moved to python 3.10 this is a good place to use a `match` statement rather than iffy code. https://docs.python.org/3/reference/compound_stmts.html#match